src/core_plugins/kibana/public/management/sections/indices/create_index_pattern_wizard/step_index_pattern/append_wildcard.js

```diff
@@ -0,0 +1,53 @@
+import { uiModules } from 'ui/modules';
+
+const module = uiModules.get('apps/management');
+
+/**
+ * This directive automatically appends a wildcard to the input field
+ * after the user starts typing. It lets the user delete the wildcard
+ * if necessary. If the value of the input field is set back to an empty
+ * string, the wildcard is immediately re-appended after the user starts
+ * typing. This is intended to be a UX improvement for the index pattern
+ * creation page. See https://github.com/elastic/kibana/pull/13454
+ */
+module.directive('appendWildcard', function () {
+  return {
+    require: 'ngModel',
+    link: function ($scope, $elem, $attrs, $ctrl) {
+      $scope.appendedWildcard = false;
+
+      function setElementValue(val, event, setCursorBackOneCharacter = false) {
+        event.preventDefault();
+        $elem.val(val);
+        if (setCursorBackOneCharacter) {
+          $elem[0].setSelectionRange(1, 1);
+        }
+      }
+
+      $elem.on('keydown', (e) => {
+        const charStr = String.fromCharCode(e.keyCode);
+        if (!/[a-z0-9]/i.test(charStr)) {
+          return;
+        }
+
+        let indexPatternName = $elem.val() + e.key;
+
+        if (indexPatternName && indexPatternName.length === 1) {
+          if (indexPatternName === '*') {
+            if ($scope.appendedWildcard) {
```

Comment:
This causes an interesting behavior, when the page first loads you're allowed to put `*` as the index-pattern, but if I create an index pattern where we append the `*` and `$scope.appendedWildcard === true`, if I try to type `*` again, it won't do anything, but if I press it again, it'll let me. This appears to be behavior that was carried over from the previous code, so if you'd prefer to create a separate PR/Issue to address this, I won't let it hold up my approval.

![asterix](https://user-images.githubusercontent.com/627123/32173105-4276bd52-bd54-11e7-881d-08f05b6d368e.gif)